32blit/engine/engine.cpp

```diff
@@ -14,9 +14,10 @@ namespace blit {
   void (*render)(uint32_t time)                     = nullptr;
 
   void set_screen_mode(ScreenMode new_mode) {
-    auto &new_screen = api.set_screen_mode(new_mode);
-    screen = Surface(new_screen.data, new_screen.format, new_screen.bounds);
-    screen.palette = new_screen.palette;
+    if(new_mode == ScreenMode::hires_palette)
+      set_screen_mode(ScreenMode::hires, PixelFormat::P);
+    else
+      set_screen_mode(new_mode, (PixelFormat)-1);
```

Comment:
I should probably define a PixelFormat::Default or something instead of the magic -1...